googlecode_wiki_to_github_markdown/googlecode_wiki_to_github_markdown.py

```diff
@@ -0,0 +1,159 @@
+    wiki = convert_headers(wiki)
+    wiki = convert_code_snippets_markers(wiki)
+    wiki = convert_comments(wiki)
+
+    wiki = remove_extra_spaces(wiki)
+    wiki = remove_extra_empty_lines(wiki)
+
+    return wiki
+
+
+def remove_gplusone(wiki):
+    return re.sub(r"<g:plusone.*</g:plusone>", "", wiki)
+
+
+def remove_toc(wiki):
+    return re.sub(r"<wiki:toc.*/>", "", wiki)
+
+
+def remove_labels(wiki):
+    return re.sub(r"#labels.*", "", wiki)
+
+
+def remove_internal_link_cancellations(wiki):
+    return wiki.replace("!" + project_name, project_name)
+
+
+def convert_internal_links(wiki):
+    for line in lines_not_in_code_snippets(wiki):
+        # [Example] -> [Example](Example.md)
+        replaced_line = re.sub(r"\[(?!http)([^ ]*)\]", r"[\1](\1.md)", line)
+        # [Example#example() link description] -> [link description](Example.md#example)
+        replaced_line = re.sub(r"\[(?!http)([^ #\(]*?)(|#[^\(]*?)(?:\(\))? +(.*?)\]", r"[\3](\1.md\2)", replaced_line)
+        if line != replaced_line:
+            wiki = wiki.replace(line, replaced_line)
+
+    return wiki
+
+
+def convert_http_links(wiki):
+    # Link only between brackets (no spaces) -> Remove brackets and ensure space after
+    wiki = re.sub(r"\[(http[^ ]*?)\] *", r"\1 ", wiki)
+    # Link with description
+    wiki = re.sub(r"\[(http.*?) +(.*?)\]", r"[\2](\1)", wiki)
+    return wiki
+
+
+def convert_code_snippets_markers(wiki):
+    wiki = re.sub(r" *\{\{\{", "\n```", wiki)
+    wiki = re.sub(r" *\}\}\}", "```\n", wiki)
+    return wiki
+
+
+def convert_comments(wiki):
+    return wiki.replace("<wiki:comment>", "<!---").replace("</wiki:comment>", "-->")
+
+
+def convert_numbered_lists(wiki):
+    for line in lines_not_in_code_snippets(wiki):
+        replaced_line = line.replace("# ", "1. ")
+        if line != replaced_line:
+            wiki = wiki.replace(line, replaced_line)
+
+    return wiki
+
+
+def convert_headers(wiki):
+    wiki = re.sub(r"^====== ?(.*[^ ]) ?======", r"###### \1", wiki, flags=re.MULTILINE)
+    wiki = re.sub(r"^===== ?(.*[^ ]) ?=====", r"##### \1", wiki, flags=re.MULTILINE)
+    wiki = re.sub(r"^==== ?(.*[^ ]) ?====", r"#### \1", wiki, flags=re.MULTILINE)
+    wiki = re.sub(r"^=== ?(.*[^ ]) ?===", r"### \1", wiki, flags=re.MULTILINE)
+    wiki = re.sub(r"^== ?(.*[^ ]) ?==", r"## \1", wiki, flags=re.MULTILINE)
+    wiki = re.sub(r"^= ?(.*[^ ]) ?=", r"# \1", wiki, flags=re.MULTILINE)
+
+    wiki = replace_summary(wiki)
+
+    return wiki
+
+
+def replace_summary(wiki):
+    """If there is a #summary tag, set all headers one level below (e.g. # -> ##) and set #summary as h1"""
+    if "#summary" in wiki:
+        wiki = wiki.replace("# ", "## ")
+        wiki = wiki.replace("#summary", '#')
+    return wiki
+
+
+def remove_extra_spaces(wiki):
+    return "\n".join(line.rstrip() for line in wiki.splitlines())
+
+
+def remove_extra_empty_lines(wiki):
+    wiki = re.sub("\n\n+", "\n\n", wiki)  # no more than 1 empty line
```

Comment:
Depends on the file (LF is only \n - mostly Linux, CRLF is \r\n - mostly Windows)